Source/Core/DolphinQt2/ToolBar.cpp

```diff
@@ -91,15 +92,19 @@ void ToolBar::MakeActions()
 
 void ToolBar::UpdateIcons()
 {
-  QString dir = Settings().GetThemeDir();
-
-  m_open_action->setIcon(QIcon(QStringLiteral("open.png").prepend(dir)));
-  m_paths_action->setIcon(QIcon(QStringLiteral("browse.png").prepend(dir)));
-  m_play_action->setIcon(QIcon(QStringLiteral("play.png").prepend(dir)));
-  m_pause_action->setIcon(QIcon(QStringLiteral("pause.png").prepend(dir)));
-  m_stop_action->setIcon(QIcon(QStringLiteral("stop.png").prepend(dir)));
-  m_fullscreen_action->setIcon(QIcon(QStringLiteral("fullscreen.png").prepend(dir)));
-  m_screenshot_action->setIcon(QIcon(QStringLiteral("screenshot.png").prepend(dir)));
-  m_config_action->setIcon(QIcon(QStringLiteral("config.png").prepend(dir)));
-  m_controllers_action->setIcon(QIcon(QStringLiteral("classic.png").prepend(dir)));
+  m_open_action->setIcon(QIcon(Resources::GetScaledPixmap("open", 4, Resources::Directory::Theme)));
```

Comment:
This isn't specific to the latest round of changes, but I didn't think of it earlier: on all desktop platforms Qt supports, this is strictly worse than how it is currently, since it won't do size mediation between the @1x and @2x versions (no current desktop is xdpi or xxdpi). Depending on the icon size needed, QIcon will fall back to the @1x versions and treat them as if they're smaller icons at the screen pixel ratio.